test/unit/org/apache/cassandra/cql3/validation/operations/CQLVectorTest.java

```diff
@@ -0,0 +1,142 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.cql3.validation.operations;
+
+import org.junit.Test;
+
+import org.apache.cassandra.cql3.CQLTester;
+
+public class CQLVectorTest extends CQLTester.InMemory
+{
+    @Test
+    public void select()
+    {
+        createTable(KEYSPACE, "CREATE TABLE %s (pk vector<int, 2> primary key)");
+
+        execute("INSERT INTO %s (pk) VALUES ([1, 2])");
+
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, 2]"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = ?", vector(1, 2)), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, 1 + 1]"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, ?]", 2), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, (int) ?]", 2), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk = [1, 1 + (int) ?]", 1), row(list(1, 2)));
+
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, 2])"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, 2], [1, 2])"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN (?)", vector(1, 2)), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, 1 + 1])"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, ?])", 2), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, (int) ?])", 2), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE pk IN ([1, 1 + (int) ?])", 1), row(list(1, 2)));
+
+        assertRows(execute("SELECT * FROM %s WHERE pk > [0, 0] AND pk < [1, 3] ALLOW FILTERING"), row(list(1, 2)));
+        assertRows(execute("SELECT * FROM %s WHERE token(pk) = token([1, 2])"), row(list(1, 2)));
+    }
+
+    @Test
+    public void insert()
+    {
+        Runnable test = () -> {
+            assertRows(execute("SELECT * FROM %s"), row(list(1, 2)));
+            execute("TRUNCATE %s");
+            assertRows(execute("SELECT * FROM %s"));
+        };
+
+        createTable(KEYSPACE, "CREATE TABLE %s (pk vector<int, 2> primary key)");
+
+        execute("INSERT INTO %s (pk) VALUES ([1, 2])");
+        test.run();
+
+        execute("INSERT INTO %s (pk) VALUES (?)", vector(1, 2));
+        test.run();
+
+        execute("INSERT INTO %s (pk) VALUES ([1, 1 + 1])");
+        test.run();
+
+        execute("INSERT INTO %s (pk) VALUES ([1, ?])", 2);
+        test.run();
+
+        execute("INSERT INTO %s (pk) VALUES ([1, (int) ?])", 2);
+        test.run();
+
+        execute("INSERT INTO %s (pk) VALUES ([1, 1 + (int) ?])", 1);
+        test.run();
+    }
+
+    @Test
+    public void insertNonPK()
```

Comment:
I think it would be useful to have a test showing that filtering works with the vector type, along the lines of:
`assertRows(execute("SELECT * FROM %s WHERE value=[1, 2] ALLOW FILTERING"), row(0, list(1, 2)));`
I have tested this and it works. I'd be quite happy if we changed this test to do this.

Reply:
ill look to add, makes sense due to this being a different code path